src/ZiggyCreatures.FusionCache/FusionCache.cs

```diff
@@ -400,7 +400,7 @@ private void CompleteBackgroundFactory<TValue>(string operationId, string key, F
 					options.ReThrowBackplaneExceptions = false;
 
 					// ADAPTIVE CACHING UPDATE
-					var lateEntry = FusionCacheMemoryEntry<TValue>.CreateFromOptions(antecedent.Result, options, false, ctx.LastModified, ctx.ETag, null);
+					var lateEntry = FusionCacheMemoryEntry<TValue>.CreateFromOptions(antecedent.GetAwaiter().GetResult(), options, false, ctx.LastModified, ctx.ETag, null);
```

Comment:
Whilst anything is better than .Result, GetAwaiter().GetResult() is still making things synchronous.

Could it not just be awaited?

Reply:
Good point, I haven't noticed that! Done.